.github/workflows/benchmark.yml

```diff
@@ -0,0 +1,105 @@
+# Run benchmark report on pull requests to master.
+# The report is added to the PR as a comment.
+
+name: Benchmarks
+
+on:
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  benchmark:
+    runs-on: ubuntu-latest
+    
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        fetch-depth: 0  # Need full history for ASV
+
+    - name: Fetch base branch
+      run: |
+        git remote add upstream https://github.com/${{ github.repository }}.git
+        git fetch upstream master
+
+    - name: Set up Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: '3.13'
+        cache: 'pip'
+
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install asv
+
+    - name: Run benchmarks
+      run: |
+        # TODO: REMOVE ONCE FIXED UPSTREAM
```

Comment:
Both `benchmark.yml` and `docs.yml` include this monkeypatch to fix https://github.com/airspeed-velocity/asv_runner/issues/45.

Unfortunately `asv` is not very polished, I had to fix 2-3 bugs to get the integration working (also see https://github.com/airspeed-velocity/asv/issues/1468). But once things are working, it's working really well.